### PR TITLE
Develop imperial

### PR DESCRIPTION
**Pull request by dalonsoa**
# Description

Combines the development done by the RSE team in different fronts and individually described in:

- #115 
- #118 
- #119 
- #121 

Mostly they refer to modernising the infrastructure of MUSE and taking the steps to easy its distribution. A couple of extra issues were open as a consequence of this. See #120 , #109 

## Type of change

Please add a line in the relevant section of
[CHANGELOG.md](https://github.com/SGIModel/StarMuse/blob/development/CHANGELOG.md) to
document the change (include PR #) - note reverse order of PR #s.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [x] All tests pass: `$ python -m pytest`
- [x] The documentation builds and looks OK: `$ python setup.py build_sphinx`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .github/workflows/broken-link-check.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+on:
+  schedule:
+    - cron: 0 0 1 * * # run monthly
+  repository_dispatch: # run manually
+    types: [check-link]
+  push:
+    branches: [main, develop_imperial]
```
Comment (dalonsoa) [inline .github/workflows/broken-link-check.yml:7]:
```suggestion
    branches: [main, develop]
```

File: setup.py
Comment (sgiarols) [inline setup.py:1]: python setup.py build_sphinx
is not working anymore

**Review by sgiarols — CHANGES_REQUESTED** [review]
It looks good. I have a few points to comment on
- the command on creating the documentation is not working anymore nor it does the command on the PR template python setup.py build_sphinx
- the README file should be updated on the use of the "muse" command rather than reporting just python -m muse ...
- fullsim[trade] fails on my laptop (MACOS, python version 3.9.0). I could see that the server does not use xlrd-1.2.0 (which I have) whereas the server environment uses more packages (
alabaster-0.7.13 ;
babel-2.12.1 ;
certifi-2023.5.7 ;
charset-normalizer-3.1.0 ;
docutils-0.18.1 ;
et-xmlfile-1.1.0 ;
imagesize-1.4.1 ;
latexcodec-2.0.1 ;
markdown-it-py-2.2.0 ;
mdit-py-plugins-0.3.5 ;
mdurl-0.1.2 ;
myst-parser-1.0.0 ;
nbsphinx-0.9.1 ;
openpyxl-3.1.2 ; 
pandoc-2.3 ;
plumbum-1.8.1 ;
ply-3.11 ;
pybtex-0.24.0 ;
pybtex-docutils-1.0.2 ;
requests-2.31.0 ;
snowballstemmer-2.2.0 ;
sphinx-5.0.2 ;
sphinxcontrib-applehelp-1.0.4 ;
sphinxcontrib-bibtex-2.5.0 ;
sphinxcontrib-devhelp-1.0.2 ;
sphinxcontrib-htmlhelp-2.0.1 ;
sphinxcontrib-jsmath-1.0.1 ;
sphinxcontrib-qthelp-1.0.3 ;
sphinxcontrib-serializinghtml-1.1.5 ;
urllib3-2.0.2 )




Regarding the broken links in the documentation, there are many which do not seem real errors. For example:
- time out messages refer to external websites, they are not real errors. 
- most of "not found" items where internal links to pages, not real errors
However, errors "404" were actual ones. For example this one https://github.com/SGIModel/MUSE_OS/blob/main/docs/tutorial-code/5-add-service-demand/1-exogenous-demand/technodata/preset/Residential2050Consumption.csv
It reminds to link to data files used for the tutorial. It bring back again the question on how far we can get the checks on this tutorial running after the model is updated

Shall we tweak a bit this error function to reduce the errors to the actual ones before updating the documentation?

File: pyproject.toml [hunk 2]
```diff
@@ -2,4 +2,128 @@
...
+authors = [
+    {name = "Sustainable Gas Institute", email = "sgi@imperial.ac.uk"},
+]
+description = "Energy System Model"
+readme = "README.md"
+requires-python = ">= 3.8, <3.10"
+keywords = ["energy", "modelling"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Programming Language :: Python :: 3.8",
```
[the remark this answers is not in the document]
Comment (dalonsoa) [reply pyproject.toml:16]: Correct. Amended in next commits.

**Comment by dalonsoa** [comment]
> fullsim[trade] fails on my laptop (MACOS, python version 3.9.0). I could see that the server does not use xlrd-1.2.0 (which I have) whereas the server environment uses more packages...

@sgiarols , could you open an issue explaining how you have run that model, error message and package versions you are using. Otherwise it is rather difficult to isolate and figure out the problem.

**Comment by dalonsoa** [comment]
Building the documentation has been fixed in 4cf01e89

The old way of doing it, `python setup.py build_sphinx`, is deprecated and Sphinx itself does not support static configuration files - it was able to use it before via `setup.py`. That's not really an issue as all the information was already in `docs/conf.py`, so it was redundant, anyway. 

I have updated the README and the pull request template to indicate the correct way of building the documentation.

**Comment by dalonsoa** [comment]
About the broken links:

- If a link to an external website fails because that website is not working, it might be just a glitch and it will work again at some point or it might be truly that the URL to the external website is no longer correct. Then is a problem in MUSE documentation and that link should be updated. For example `https://www.anaconda.com/what-is-anaconda/` really does not exist and needs to be updated. Also, for papers you should use proper DOIs rather than URLs, so https://www.sciencedirect.com/science/article/pii/S0306261920308072 should be included as https://doi.org/10.1016/j.apenergy.2020.115295, which is guaranteed to exist (by the definition of DOI, as far as I can tell).
- The internal links are also related to locations that do not exist. For example, if you go to https://muse-os.readthedocs.io/en/latest/source/muse.outputs.html#muse.outputs.YearlyAggregate, it indeed open the page, but does not go to the tag `#muse.outputs.YearlyAggregate`, as required, because that tag does not exist. What does exist is `#muse.outputs.sinks.YearlyAggregate`. I guess that at some point the location of that class was changed, but the documentation was not. And the same with the others.

So, I don't see why they should not be flagged as errors. 

About the errors related to missing files I'm not totally sure what you mean. Could you elaborate?

I'm going to investigate the options of the tool to see if it can tell us where the broken link is located, not just that there is a broken link, which will make sorting them out much easier.

**Comment by dalonsoa** [comment]
It is not possible to report the location of the broken link, at the moment.[ I've created an issue in their repo](https://github.com/ruzickap/action-my-broken-link-checker/issues/104) to see if that can be added. For now, we need to review the documentation manually - using the `search` function in VSCode, obviously.

**Comment by dalonsoa** [comment]
About updating the way of using MUSE in the readme, I will do it once we have finish #106 , so we provide an even simpler way of doing things. For now, the existing instructions are still valid.

**Comment by dalonsoa** [comment]
@sgiarols , I've added the automatic deployment of MUSE to PyPI. It has taken a few attempts to get the configuration right, but it is working, now. The instructions in the README will not work until we push a new tag to PyPI - for now, [it is only in TestPyPI](https://test.pypi.org/project/MUSE-OS/), which is not fit for production, only for testing, as the name suggests. 

I will clean the history and clear the tags once you have reviewed and approved the PR - but before merging - and we will make a proper release as MUSE 1.1.0 once this PR is merged.

**Review by sgiarols — APPROVED** [review]

**Comment by dalonsoa** [comment]
@sgiarols , I've clean the history and removed the unnecessary tags, so this is ready for merging.

**Comment by sgiarols** [comment]
> > fullsim[trade] fails on my laptop (MACOS, python version 3.9.0). I could see that the server does not use xlrd-1.2.0 (which I have) whereas the server environment uses more packages...
> 
> @sgiarols , could you open an issue explaining how you have run that model, error message and package versions you are using. Otherwise it is rather difficult to isolate and figure out the problem.

done here #127

**Pull request merged**